Sources/SwiftRefactor/ConvertComputedPropertyToZeroParameterFunction.swift

```diff
@@ -0,0 +1,77 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=6)
+public import SwiftSyntax
+#else
+import SwiftSyntax
+#endif
+
+public struct ConvertComputedPropertyToZeroParameterFunction: SyntaxRefactoringProvider {
+  public static func refactor(syntax: VariableDeclSyntax, in context: Void) -> FunctionDeclSyntax? {
+    guard syntax.bindings.count == 1,
+      let binding = syntax.bindings.first,
+      var identifierPattern = binding.pattern.as(IdentifierPatternSyntax.self)
+    else { return nil }
+
+    identifierPattern.leadingTrivia = syntax.bindingSpecifier.trailingTrivia
+
+    guard let typeAnnotation = binding.typeAnnotation,
+      let accessorBlock = binding.accessorBlock,
+      case let .getter(statements) = accessorBlock.accessors
+    else { return nil }
+
+    let returnType = typeAnnotation.type
+
+    var returnClause: ReturnClauseSyntax?
+    var trailingTrivia: Trivia = []
+
+    if !returnType.isVoid {
+      let trailingTrivia = typeAnnotation.colon.leadingTrivia + identifierPattern.identifier.trailingTrivia
+      returnClause = ReturnClauseSyntax(
+        arrow: .arrowToken(
+          leadingTrivia: trailingTrivia.isEmpty ? .space : trailingTrivia,
+          trailingTrivia: typeAnnotation.colon.trailingTrivia
+        ),
+        type: returnType
+      )
+    } else {
+      trailingTrivia = typeAnnotation.colon.trailingTrivia
+    }
+
+    let body = CodeBlockSyntax(
+      leftBrace: accessorBlock.leftBrace,
+      statements: statements,
+      rightBrace: accessorBlock.rightBrace
+    )
+
+    let functionSignature = FunctionSignatureSyntax(
+      parameterClause: FunctionParameterClauseSyntax(parameters: []),
+      returnClause: returnClause,
+      trailingTrivia: trailingTrivia
+    )
```

Comment:
Why do we need to add the space as leading trivia to the arrow? Wouldn’t it be easier if the `)` in the parameter clause always has a trailing space trivia? That would also match how the parser would qualify the trivia.

The arrow token could then just inherit the trivia from the colon. What I would do is something like the following

```suggestion
    let triviaAfterSignature: Trivia

    if !returnType.isVoid {
      triviaAfterSignature = .space
      returnClause = ReturnClauseSyntax(
        arrow: .arrowToken(
          leadingTrivia: typeAnnotation.colon.leadingTrivia,
          trailingTrivia: typeAnnotation.colon.trailingTrivia
        ),
        type: returnType
      )
    } else {
      triviaAfterSignature = typeAnnotation.colon.leadingTrivia + typeAnnotation.colon.trailingTrivia
    }

    let body = CodeBlockSyntax(
      leftBrace: accessorBlock.leftBrace,
      statements: statements,
      rightBrace: accessorBlock.rightBrace
    )

    let functionSignature = FunctionSignatureSyntax(
      parameterClause: FunctionParameterClauseSyntax(parameters: []),
      returnClause: returnClause,
      trailingTrivia: identifierPattern.identifier.trailingTrivia + triviaAfterSignature
    )
```